app/src/main/java/org/schabi/newpipe/download/DeleteManager.java

```diff
@@ -0,0 +1,178 @@
+package org.schabi.newpipe.download;
+
+import android.app.Activity;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.design.widget.BaseTransientBottomBar;
+import android.support.design.widget.Snackbar;
+import android.view.View;
+
+import org.schabi.newpipe.R;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import io.reactivex.Completable;
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import io.reactivex.subjects.PublishSubject;
+import us.shandian.giga.get.DownloadManager;
+import us.shandian.giga.get.DownloadMission;
+
+public class DeleteManager {
+
+    private static final String KEY_STATE = "delete_manager_state";
+
+    private View mView;
+    private HashSet<String> mPendingMap;
+    private List<Disposable> mDisposableList;
+    private DownloadManager mDownloadManager;
+    private PublishSubject<DownloadMission> publishSubject = PublishSubject.create();
+
+    DeleteManager(Activity activity) {
+        mPendingMap = new HashSet<>();
+        mDisposableList = new ArrayList<>();
+        mView = activity.findViewById(android.R.id.content);
+    }
+
+    public Observable<DownloadMission> getUndoObservable() {
+        return publishSubject;
+    }
+
+    public boolean contains(@NonNull DownloadMission mission) {
+        return mPendingMap.contains(mission.url);
+    }
+
+    public void add(@NonNull DownloadMission mission) {
+        mPendingMap.add(mission.url);
+
+        if (mPendingMap.size() == 1) {
+            showUndoDeleteSnackbar(mission);
+        }
+    }
+
+    public void setDownloadManager(@NonNull DownloadManager downloadManager) {
+        mDownloadManager = downloadManager;
+
+        if (mPendingMap.size() < 1) {
+            //nothing to do
```

Comment:
I think it's clear from the code that you want to abord the call here, so no need to commend it.

You could shorten it like this: `if(mPendingMap.size() < 1) return;`.